julia/mmtk_julia.c

```diff
@@ -46,73 +46,29 @@ JL_DLLEXPORT void (jl_mmtk_harness_end)(void)
     mmtk_harness_end();
 }
 
-JL_DLLEXPORT jl_value_t *jl_mmtk_gc_alloc_default(jl_ptls_t ptls, int osize, size_t align, void *ty)
+static void jl_gc_free_memory(jl_value_t *v, int isaligned) JL_NOTSAFEPOINT
```

Comment:
Why do we need to implement this function? Did Julia remvove the function? If so, why do we still need it? Is this implementation based on any function in Julia?

Reply:
Julia has its own version. As I mentioned in the comment I've added, it is slightly different because Stock needs to update the `gc_heap_stats` variable which is not used by MMTk.

Reply:
Where are the allocation functions for those objects? Are the functions in the Julia repo? Are they in gc-common.c, or specific GC impls?

It is a bit weird that we only see `free` here, but not the allocation. The allocation and the free should be paired, right? They both should be in gc-common, or in each specific GC impl. But anyway, I am okay if we leave it as it is and see if the Julia guys are okay with this.

Reply:
I'm trying to move everything to `gc-mmtk.c`, I've moved the allocation functions already and left only the functions related to `Julia_Upcalls`. But I'm trying to define them as `extern C` instead so we can move everything. Will update this PR when I'm done.